config/routes/api/v2.rb

```diff
@@ -1,6 +1,7 @@
 # config/routes/api/v2.rb
 Foreman::Application.routes.draw do
   namespace :api, :defaults => {:format => 'json'} do
+    puppet_plugin = Foreman::Plugin.find(:foreman_puppet_enc)
```

Comment:
Isn't it foreman_puppet:
```suggestion
    puppet_plugin = Foreman::Plugin.find(:foreman_puppet)
```